src/services/userAuth.js

```diff
@@ -0,0 +1,97 @@
+import bcrypt from 'bcrypt';
+import jwt from 'jsonwebtoken';
+import uuid from 'uuid/v1';
+import dotenv from 'dotenv';
+
+import database from '../models/index';
+import doesUserexist from '../helpers/userHelper';
+import notification from './notification';
+
+dotenv.config();
+
+const saltRounds = 10;
+const users = database.User;
+
+/**
+ * Signs in a user
+ * @param {Object} request - The request object to the server
+ * @param {Object} response - The response object from the server
+ * @param {Function} Next - Calls the express error handler
+ * @return {Void} - Returns nothing
+ */
+export const createUser = async (request, response, Next) => {
+  const userPresent = await doesUserexist(request.body.email);
+  if (!userPresent) {
+    const hash = await bcrypt.hashSync(request.body.password, saltRounds);
+    const userData = { ...request.body, password: hash, id: uuid() };
+
+    const createdUser = await users.create(userData);
+    const { password, ...newUser } = createdUser.dataValues;
+    const host = request.get('host');
+    notification.emit('notification', {
+      type: 'accountActivation',
+      payload: [{
+        email: newUser.email,
+        name: newUser.firstName,
+        verificationLink: `https://${host}/profile`
+      }]
+    });
+
+    const token = jwt.sign(
+      { id: createdUser.dataValues.id },
+      process.env.SECRET_KEY,
+      { expiresIn: '14d' }
+    );
+
+    response.status(200).json({ status: 200, user: newUser, token });
+  } else {
+    const err = new Error();
+    err.message = 'you are already registered';
+    err.status = 409;
+    Next(err);
+  }
+};
+
+/**
+ * Creates a user
+ * @param {Object} request - The request object to the server
+ * @param {Object} response - The response object from the server
+ * @param {Function} Next - Calls the express error handler
+ * @return {Void} - Returns nothing
+ */
+export const checkinUser = async (request, response, Next) => {
+  const userData = { ...request.body };
+  const createdUser = await users.findAll({
```

Comment:
Can you use `findOne` instead of `findAll`, since the email is generically unique it is expected that only one of such email exists in the database.

The  `findOne` method also returns a Sequelize object, this will help in dealing with the array issue.

Reply:
ya, was supposed to change from 'all' to 'one'